src/core/tranche/assessor.sol

```diff
@@ -0,0 +1,78 @@
+// Copyright (C) 2019 Centrifuge
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+
+pragma solidity >=0.4.24;
+
+contract OperatorLike {
+    function balance() public returns(uint);
+    function debt() public returns(uint);
+}
+
+contract TrancheManagerLike {
+    function trancheCount() public returns(uint);
+    function isJunior(address) public returns(bool);
+    function poolValue() public returns(uint);
+    function indexOf(address) public returns(int);
+    function juniorOperator() public returns(address);
+    function seniorOperator() public returns(address);
+}
+
+contract Assessor {
+    // --- Data ---
+    TrancheManagerLike trancheManager; 
+
+    // --- Assessor ---
+    // computes the current asset value for tranches.
+    constructor(address trancheManager_) public {
+        trancheManager = TrancheManagerLike(trancheManager_);
+    }
+
+    function getAssetValueFor(address operator_) public returns (uint) {
+         OperatorLike operator = OperatorLike(operator_);
+         uint trancheDebt = operator.debt();
+         uint trancheReserve = operator.balance();
+         uint poolValue = trancheManager.poolValue();
+         if (trancheManager.isJunior(operator_)) {
```

Comment:
I think maybe the trancheManager shouldn't know that we are comparing operator addresses here.

why not just:
```javascript
if(operator_ == trancheManager.juniorTranche())
```